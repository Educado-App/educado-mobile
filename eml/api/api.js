import axios from 'axios';

const timeoutInMs = 1200;

const url = 'https://educado-backend-staging-x7rgvjso4a-ew.a.run.app/';// Change this to your LOCAL IP address when testing.

const certificateUrl = 'http://172.30.245.212:8080';

/* Commented out for avoiding linting errors :))
 * TODO: move IP address to .env file !!!
const testUrl = 'http://localhost:8888';
const testExpo = 'http://172.30.211.57:8888'; 
const digitalOcean = 'http://207.154.213.68:8888';
*/

/*** COURSE, SECTIONS AND EXERCISES ***/

// Get components for a specific section
export const getComponents = async (sectionId) => {
	try {
		const res = await axios.get(url + '/api/courses/sections/' + sectionId + '/components');
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

export const getSectionById = async (sectionId) => {
	try {
		const res = await axios.get(url + '/api/sections/' + sectionId);
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}

	}
};

// Get specific course

export const getCourse = async (courseId) => {
	try {
		const res = await axios.get(url + '/api/courses/' + courseId, {timeout: timeoutInMs});
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

// Get all courses
export const getCourses = async () => {
	try {
		const res = await axios.get(url + '/api/courses');
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

// Get all sections for a specific course
export const getAllSections = async (courseId) => {
	try {
		const res = await axios.get(url + '/api/courses/' + courseId + '/sections', {timeout: timeoutInMs});
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

// Get specific section
// ************* same as getSectionById *************
export const getSection = async (courseId, sectionId) => {
	try {
		const res = await axios.get(
			url + '/api/courses/' + courseId + '/sections/' + sectionId
		);
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};


// Get all lectures in a specific section:
export const getLecturesInSection = async (sectionId) => {
	try {
		const res = await axios.get(
			url + '/api/lectures/section/' + sectionId
			, {timeout: timeoutInMs});
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

/*** SUBSCRIPTION ***/

// Get user subscriptions
export const getSubscriptions = async (userId) => {
	try {
		// maybe not best practise to pass user ID as request query
		// but this is the only format where it works
		// passing user ID as request body for get request gives error
		const res = await axios.get(
			url + '/api/students/' + userId + '/subscriptions',
			{timeout: 1200});

		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

// Subscribe to course
export const subscribeToCourse = async (userId, courseId) => {
	try {
		await axios.post(
			url + '/api/courses/' + courseId + '/subscribe',
			{
				user_id: userId,
			}
		);
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

// Unsubscribe to course
export const unSubscribeToCourse = async (userId, courseId) => {
	try {
		await axios.post(
			url + '/api/courses/' + courseId + '/unsubscribe',
			{
				user_id: userId,
			}
		);
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};


// Get certificates from student
export const fetchCertificates = async (userId) => {
	try {
		if (userId == null) {
			throw 'User ID is null';
		}
		const res = await axios.get(certificateUrl + '/api/student-certificates/student/' + userId);
		return res.data;
	}  catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

//CREATED BY VIDEO STREAM TEAM
/*This will be improved in next pull request to handle getting different resolutions properly 
with our new video streaming service in go.
*/

export const getVideoStreamUrl = (fileName, resolution) => {

	let resolutionPostfix;
	switch (resolution) {
	case '360':
		resolutionPostfix = '_360x640';
		break;
	case '480':
		resolutionPostfix = '_480x854';
		break;
	case '720':
		resolutionPostfix = '_720x1280';
		break;
	case '1080':
		resolutionPostfix = '_1080x1920';
		break;
	default:
		resolutionPostfix = '_360x640';
	}

	return `${url}/api/bucket/stream/${fileName}${resolutionPostfix}.mp4`;
};

export const getLectureById = async (lectureId) => {
	try {
		const res = await axios.get(url + '/api/lectures/' + lectureId);
		return res.data;
	} catch (err) {
		if (err?.response?.data != null) {
			throw err.response.data;
		} else {
			throw err;
		}
	}

};


export const getBucketImage = async (fileName) => {
	try {
		const res = await axios.get(
			`${url}/api/bucket/${fileName}`
			, {timeout: timeoutInMs});
		return `data:image/png;base64,${res.data}`;
	} catch (err) {
		if (err?.response?.data != null) {
			throw err.response.data;
		} else {
			throw err;
		}
	}
};

