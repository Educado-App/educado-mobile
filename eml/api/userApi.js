import axios from 'axios';

/* Commented out to avoid linting errors 
 * TODO: move IP address to .env file !!!
const prod = 'http://educado.somethingnew.dk';
const test = 'http://172.30.211.110:8888'; // Change this to your LOCAL IP address when testing.
const local = 'http://localhost:8888';
const digitalOcean = 'http://207.154.213.68:8888';
*/ 

const url = 'https://educado-backend-staging-x7rgvjso4a-ew.a.run.app/';// Change this to your LOCAL IP address when testing.

/**
 * This is the client that will be used to make requests to the backend.
 */
export const client = axios.create({
	baseURL: url,
	withCredentials: true,
	responseType: 'json',
	timeout: 30000,
});

/**
 * Sends a request to the backend to register a new user.
 * @param {Object} obj Should contain the following properties:
 * - firstName
 * - lastName
 * - email
 * - password
 */
export const registerUser = async (obj) => {
	console.log(`User trying to register:
    firstName: ${obj.firstName ?? 'undefined'}
    lastName: ${obj.lastName ?? 'undefined'}
    email: ${obj.email ?? 'undefined'}`);

	try {
		const res = await client.post('/api/auth/signup', obj);
		console.log('User successfully registered');
		return res.data;
	} catch (e) {
		console.log(e.message);
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

/**
 * Sends a request to the backend to log in an existing user.
 * @param {Object} obj should contain the following properties:
 * - email
 * - password
 */
export const loginUser = async (obj) => {
	try {
		const res = await client.post('/api/auth/login', obj);
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

export const deleteUser = async (user_id, token) => {
	try {
		const res = await client.delete('/api/users/' + user_id, {
			headers: {
				'Content-Type': 'application/json',
				'token': token, // Include the token in the headers
			},
		});
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

export const updateUserFields = async (user_id, obj, token) => {
	try {
		const res = await client.patch(`/api/users/${user_id}`, obj, {
			headers: {
				'Content-Type': 'application/json',
				'token': token, // Include the token in the headers
			},
		});
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

export const updateUserPassword = async (user_id, oldPassword, newPassword, token) => {
	try {
		const res = await axios.patch(url + `/api/users/${user_id}/password`, {
			oldPassword: oldPassword,
			newPassword: newPassword,
		}, {
			headers: {
				'Content-Type': 'application/json',
				'token': token, // Include the token in the headers
			},
		});

		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

export const completeComponent = async (user_id, comp, isComplete, points, token) => {
	try{
		const res = await client.patch('/api/students/' + user_id + '/complete', { comp: comp, isComplete: isComplete, points: points }, {
			headers: {
				'Content-Type': 'application/json',
				'token': token, // Include the token in the headers
			},
		});

		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};

export const getStudentInfo = async (user_Id) => {
	try {
		const res = await client.get('/api/students/' + user_Id + '/info');
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};



export const addCourseToStudent = async (user_Id, course_Id, token) => {
	try {
		const res = await client.patch('/api/students/' + user_Id + '/courses/' + course_Id + '/enroll',
			{},
			{
				headers: {
					'Content-Type': 'application/json',
					'token': token, // Include the token in the headers
				},
			}
		);

		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};



/**
 * Function to send mail to user with code to reset password
 * @param {Object} email should contain an email, to receive a reset password message
*/
export const sendResetPasswordEmail = async (email) => {
	try {
		const res = await client.post('/api/auth/reset-password-request', email);
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};


/**
 * function to validate the code sent to the user 
 * @param {Object} obj should contain the following properties:
 * - email
 * - token
*/
export const validateResetPasswordCode = async (obj) => {
	try {
		const res = await client.post('/api/auth/reset-password-code', obj);
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}

};

/**
 * When user enters a new password it should update the password of the user
 * @param {Object} obj should contain the following properties:
 * - email
 * - token
 * - newPassword
*/
export const enterNewPassword = async (obj) => {
	try {
		const res = await client.patch('/api/auth/reset-password', obj);
		return res.data;
	} catch (e) {
		if (e?.response?.data != null) {
			throw e.response.data;
		} else {
			throw e;
		}
	}
};