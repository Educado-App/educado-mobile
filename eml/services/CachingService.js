let type = navigator.connection.effectiveType;

function updateConnectionStatus() {
  console.log(`Connection type changed from ${type} to ${navigator.connection.effectiveType}`);
  type = navigator.connection.effectiveType;
}

navigator.connection.addEventListener('change', updateConnectionStatus);

let downlaodCourse = false;
const connection = navigator.connection;
if (connection) {
  if (connection.effectiveType === 'wifi') {
    downloadCourse = true;
  }
}