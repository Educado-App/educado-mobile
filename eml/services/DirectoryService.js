import { React } from 'react'
import * as FileSystem from 'expo-file-system';

export async function CreateDirectory(name) {

    //Create a Directory by the (name)
    if (name !== undefined){

        try {
            await FileSystem.makeDirectoryAsync(FileSystem.documentDirectory + name)
                .then( () =>{
                        console.log("Created directory: " + name);
                        return name;
                    }
                )
                .catch(error => {
                    console.log(error)
                });
        }
        catch (error){
            console.log(error);
        }

    } else console.log("error: directory must have a name!");

}
export async function ReadDirectory(name){

        //Returns an ARRAY containing the names of existing items in the directory (name)

        let items;

        try {
             await FileSystem.readDirectoryAsync(FileSystem.documentDirectory + name)
                .then(filesInDirectory => {
                    items = filesInDirectory;
                    console.log("ITEMS: ", items);
                })
                .catch(error => {
                    console.log("Error Reading directory (maybe it is not yet created)");
                })
        }
        catch (error){
            console.log(error);
        }

        return items;

    }
export async function DeleteDirectory(name){

    //Delete the directory located by name

    try {
         await FileSystem.deleteAsync(FileSystem.documentDirectory + name)
            .then(() => {
                console.log("Deleted directory " + name);
            })
            .catch(error => {
                console.log("Error deleting the directory. (maybe it doesn't exist)");
            });
    }
    catch (error){
        console.log("error");
    }

}

export async function DownloadAndStoreContent(url, directory, name){

        //DOWNLOAD the video from the 'url' and STORE it in the 'directory'
        //Returns the local uri to the file
        //The name of the file is a substring of the url from the last '/'
        //For example : http://example.com/video.mp4 will be named 'video.mp4'

        const localUri = FileSystem.documentDirectory + directory + '/' + name;

        try {
              return await FileSystem.downloadAsync(url, localUri)
                .then(({ uri }) => {
                    console.log('Finished downloading to', uri);
                    return localUri;
                })
                .catch(error => {
                    console.error(error);
                });
        }
        catch (error){
            console.log(error);
        }
    }
export async function DeleteVideoByUri(uri) {

        //Delete the video located in the uri

        try {
             await FileSystem.deleteAsync(uri)
                .then(() => {
                    console.log(uri.substring(uri.lastIndexOf('/') + 1) + " Successfully Deleted!");
                })
                .catch(error => {
                    console.log(error);
                });
        }
        catch (error){

            console.log(error);
        }
    }
export async function DeleteVideoByName(name, directory) {

        //Delete the video located in the local directory by its name

        let localName = FileSystem.documentDirectory + directory + '/' + name;

        try {
             await FileSystem.deleteAsync(localName)
                .then(() => {
                    console.log(name + " deleted!");
                })
                .catch(error => {
                    console.log(error);
                });
        }
        catch (error){
            console.log(error);
        }
    }




