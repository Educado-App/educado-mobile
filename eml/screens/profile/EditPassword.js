import React, { useEffect, useState } from 'react';

// Components
import { View,SafeAreaView, } from 'react-native';
import PasswordField from '../../components/general/forms/PasswordField';
import FormButton from '../../components/general/forms/FormButton';
import ToastNotification from '../../components/general/ToastNotification';
import BackButton from '../../components/general/BackButton';
import { useNavigation } from '@react-navigation/native';
import Text from '../../components/general/Text';
// Services
import { updateUserPassword } from '../../api/userApi';
import { getUserInfo, getJWT } from '../../services/StorageService';
import FormFieldAlert from '../../components/general/forms/FormFieldAlert';
import { validatePasswordContainsLetter, validatePasswordLength } from '../../components/general/Validation';
import { alertErrorCode } from '../../services/ErrorAlertService';
import BaseScreen from '../../components/general/BaseScreen';

/**
 * Edit password screen
 * @returns {React.Element} Component for the edit profile screen
 */
export default function EditPassword() {
	// States
	const [currentPassword, setCurrentPassword] = useState('');
	const [newPassword, setNewPassword] = useState('');
	const [confirmPassword, setConfirmPassword] = useState('');
	const [isFormFilledOut, setIsFormFilledOut] = useState(false);
	const [passwordAlert, setPasswordAlert] = useState('');
	const navigation = useNavigation();

	// password validation
	useEffect(() => {
		// reset alert if no new passsword has been entered
		if (newPassword.length == 0) return setPasswordAlert('');

		// password must contain at least one letter
	 	if (!validatePasswordContainsLetter(newPassword)) {
	 		return setPasswordAlert('Senha deve conter pelo menos 1 letra');
	 	}

		// validating password length (at least 8 characters)
	 	if (!validatePasswordLength(newPassword)) {
	 		return setPasswordAlert('Senha muito curta! Mínimo 8 caracteres');
	 	}

		// newPassword and confirmPassword must match
	 	if (newPassword !== confirmPassword) {
	 		return setPasswordAlert('As senhas não coincidem!');
	 	}

	 	setPasswordAlert('');
	 }, [newPassword, confirmPassword]);

	useEffect(() => {
		setIsFormFilledOut(checkIsFormFilledOut());
	}, [currentPassword, newPassword, confirmPassword, passwordAlert]);

	// Gauge whether the form is filled out
	const checkIsFormFilledOut = () => {
		if(currentPassword.length == 0) return false;
		if(newPassword.length == 0) return false;
		if(confirmPassword.length == 0) return false;
		if(passwordAlert != '') return false;
		return true;
	};

	// Submit password change
	const submitForm = async () => {
		try {
			// Get user ID
			const userInfo = await getUserInfo();
			const JWT = await getJWT();
			// Make request
			await updateUserPassword(userInfo.id, currentPassword, newPassword, JWT);
			// Show success message
			ToastNotification('success', 'Senha alterada!');
				
			// Clear fields
			setCurrentPassword('');
			setNewPassword('');
			setConfirmPassword('');

			// reset error state
			setPasswordAlert('');
			
			// navigation.navigate('Perfil');
		} catch(error) {
			// Handle errors
			console.log('Error updating password:',error?.data?.error ?? error);
			
			switch (error?.error?.code) {
				case 'E0806':
					// Your password is incorrect!
					console.log('Your password is incorrect!');
					setPasswordAlert('Senha atual está incorreta!');
					break;
					// TODO: What error should we give here instead? Unknown error? 
				default: // Errors not currently handled with specific alerts
					ShowAlert('Erro desconhecido!');
				}
		}
	};
	return (
		<BaseScreen>
			<SafeAreaView className='bg-secondary'>
				<View className='flex flex-col mx-4 z-10'>
					<View>
						<View className='relative mt-12 mb-6'>
							{/* Back button */}
							<BackButton onPress={() => navigation.navigate('Perfil')} />

							{/* Title */}
							<Text className='w-full text-center text-xl font-sans-bold'>
						Alterar senha
							</Text>
						</View>
					</View>
					<PasswordField
						label='Senha atual'
						password={currentPassword}
						setPassword={setCurrentPassword}
						className='mb-4'
						/>
					<PasswordField
						label='Nova senha'
						password={newPassword}
						setPassword={setNewPassword}
						className='mb-4'
						/>
					<PasswordField
						label='Confirmar senha'
						password={confirmPassword}
						setPassword={setConfirmPassword}
						/>
					<FormFieldAlert testId="passwordAlert" label={passwordAlert} />
					<FormButton
						onPress={submitForm}
						disabled={!isFormFilledOut}
						className='mt-4'
						>
						Confirmar
					</FormButton>
				</View>
			</SafeAreaView>
		</BaseScreen>
	);
}