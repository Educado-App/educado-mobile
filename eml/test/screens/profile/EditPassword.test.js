import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import EditPassword from '../../../screens/profile/EditPassword'; // Adjust this import according to the actual location
import { updateUserPassword } from '../../../api/userApi';
import { getUserInfo, getJWT } from '../../../services/StorageService';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

jest.mock('../../../api/userApi', () => ({
  updateUserPassword: jest.fn(),
}));

jest.mock('../../../services/StorageService', () => ({
  getUserInfo: jest.fn(),
  getJWT: jest.fn(),
}));

describe('EditPassword Component', () => {
  const mockUserInfo = { id: '12345' };
  const mockJWT = 'mockJWTToken';

  beforeEach(() => {
    getUserInfo.mockResolvedValue(mockUserInfo);
    getJWT.mockResolvedValue(mockJWT);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { getByText } = render(<EditPassword />);
    expect(getByText('Alterar senha')).toBeTruthy();
  });
});
