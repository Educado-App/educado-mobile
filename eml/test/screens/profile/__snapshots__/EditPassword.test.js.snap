// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPassword Component renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    [
      {
        "backgroundColor": "#F1F9FB",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "display": "flex",
        },
        {
          "flexDirection": "column",
        },
        {
          "marginLeft": 16,
          "marginRight": 16,
        },
        {
          "zIndex": 10,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "position": "relative",
            },
            {
              "marginTop": 48,
            },
            {
              "marginBottom": 24,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "position": "absolute",
              },
              {
                "zIndex": 50,
              },
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "marginBottom": -16,
                "marginLeft": -16,
                "marginRight": -16,
                "marginTop": -16,
              },
            ]
          }
        >
          <View
            style={
              {
                "overflow": "hidden",
              }
            }
            testID="RNE__ICON__CONTAINER"
          >
            <View
              accessibilityRole="button"
              android_ripple={
                {
                  "borderless": false,
                  "color": "rgba(0, 0, 0, 0.3)",
                  "radius": -5,
                }
              }
              disabled={false}
              testID="RNE__ICON__CONTAINER_ACTION"
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                testID="RNE__ICON"
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "width": "100%",
                },
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
                {
                  "fontFamily": "Montserrat-Bold",
                },
              ],
            ]
          }
        >
          Alterar senha
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "position": "relative",
          },
          [
            {
              "marginBottom": 16,
            },
          ],
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 8,
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                ],
              ]
            }
          >
            Senha atual
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 4,
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#FF4949",
                  },
                ],
              ]
            }
          >
            *
          </Text>
        </View>
        <View>
          <TextInput
            autoComplete="off"
            keyboardType="default"
            onChangeText={[Function]}
            passwordGuidelines={false}
            placeholder="Insira sua senha"
            secureTextEntry={true}
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "paddingBottom": 4,
                  "paddingTop": 4,
                },
                {
                  "paddingLeft": 10,
                },
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                },
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "position": "absolute",
            },
            {
              "right": 0,
            },
            {
              "top": 20,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#FF4949",
              },
              {
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "position": "relative",
          },
          [
            {
              "marginBottom": 16,
            },
          ],
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 8,
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                ],
              ]
            }
          >
            Nova senha
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 4,
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#FF4949",
                  },
                ],
              ]
            }
          >
            *
          </Text>
        </View>
        <View>
          <TextInput
            autoComplete="off"
            keyboardType="default"
            onChangeText={[Function]}
            passwordGuidelines={false}
            placeholder="Insira sua senha"
            secureTextEntry={true}
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "paddingBottom": 4,
                  "paddingTop": 4,
                },
                {
                  "paddingLeft": 10,
                },
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                },
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "position": "absolute",
            },
            {
              "right": 0,
            },
            {
              "top": 20,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#FF4949",
              },
              {
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "position": "relative",
          },
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 8,
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                ],
              ]
            }
          >
            Confirmar senha
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "color": "#383838",
                },
                {
                  "fontSize": 16,
                },
                [
                  {
                    "marginLeft": 4,
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                  {
                    "color": "#FF4949",
                  },
                ],
              ]
            }
          >
            *
          </Text>
        </View>
        <View>
          <TextInput
            autoComplete="off"
            keyboardType="default"
            onChangeText={[Function]}
            passwordGuidelines={false}
            placeholder="Insira sua senha"
            secureTextEntry={true}
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                {
                  "paddingBottom": 4,
                  "paddingTop": 4,
                },
                {
                  "paddingLeft": 10,
                },
                {
                  "backgroundColor": "#FFFFFF",
                },
                {
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                },
              ]
            }
            value=""
          />
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "position": "absolute",
            },
            {
              "right": 0,
            },
            {
              "top": 20,
            },
            {
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Text />
      </View>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            {
              "color": "#383838",
            },
            {
              "fontSize": 16,
            },
            [
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#FF4949",
              },
              {
                "marginLeft": 8,
                "marginRight": 8,
              },
            ],
          ]
        }
      />
    </View>
    <View>
      <View
        accessibilityState={
          {
            "disabled": true,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#166276",
            "borderBottomLeftRadius": 8,
            "borderBottomRightRadius": 8,
            "borderTopLeftRadius": 8,
            "borderTopRightRadius": 8,
            "marginTop": 16,
            "opacity": 0.5,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              {
                "color": "#383838",
              },
              {
                "fontSize": 16,
              },
              [
                {
                  "textAlign": "center",
                },
                {
                  "fontFamily": "Montserrat-Bold",
                },
                {
                  "fontSize": 16,
                },
                {
                  "color": "#FFFFFF",
                },
              ],
            ]
          }
        >
          Confirmar
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
